e2e/cypress/integration/spatial.cy.js

```diff
@@ -0,0 +1,69 @@
+const group = {
+    name: 'climate',
+    desc: 'Climate Group'
+}
+describe('Spatial', { testIsolation: false }, () => {
+    before(() => {
+        cy.login();
+        cy.create_group(group.name, group.desc);        
+    });
+
+    after(() => {
+        cy.delete_group(group.name);
+        cy.logout();
+    });
+
+    it('Can search geographies by name', () => {
+        cy.request('/api/3/action/location_search?q=california').should((response) => {
+            expect(response.body).to.have.property('success', true);
+            expect(response.body.result[0]).to.have.property('text', 'California');
+        });
+    });
+
+    it('The map view works and can draw box and search', () => {
+        cy.visit('/dataset');
+        cy.get('.leaflet-control-custom-button').click();
+        cy.get('.modal-spatial-query .modal-footer').find('.disabled');
+        cy.get('#draw-map-container')
+            .trigger('mousedown', { which: 1 })
+            .trigger('mousemove', { clientX: 500, clientY: 153 })
+            .trigger('mouseup');
+        cy.hide_debug_toolbar();
+        // click the apply button then on the next redirected page find the box
+        // on the map and content in the body
+        cy.get('.modal-spatial-query .modal-footer').find('[class="btn btn-primary apply"]').click();
+        cy.get('#dataset-map-container').find('svg.leaflet-zoom-animated');
+        cy.contains(/No datasets found/);
```

Comment:
I would like to add a complete working test here, maybe adding a dataset with spatial data so that the spatial search returns a dataset?

Reply:
+1 on this enhancement